service/worker/deployment/deployment_workflow.go

```diff
@@ -346,6 +329,89 @@ func (d *DeploymentWorkflowRunner) handleSyncState(ctx workflow.Context, args *d
 	}, nil
 }
 
+func (d *DeploymentWorkflowRunner) syncToTaskQueues1(ctx workflow.Context) error {
+	syncReq := &deploymentspb.SyncUserDataRequest{
+		Deployment: d.State.Deployment,
+	}
+	for tqName, byType := range d.State.TaskQueueFamilies {
+		for tqType, data := range byType.TaskQueues {
+			syncReq.Sync = append(syncReq.Sync, &deploymentspb.SyncUserDataRequest_SyncUserData{
+				Name: tqName,
+				Type: enumspb.TaskQueueType(tqType),
+				Data: d.dataWithTime(data),
+			})
+		}
+	}
+	activityCtx := workflow.WithActivityOptions(ctx, defaultActivityOptions)
+	var syncRes deploymentspb.SyncUserDataResponse
+	err := workflow.ExecuteActivity(activityCtx, d.a.SyncUserData, syncReq).Get(ctx, &syncRes)
+	if err != nil {
+		return err
+	}
+	if len(syncRes.TaskQueueMaxVersions) > 0 {
+		// wait for propagation
+		checkReq := &deploymentspb.CheckUserDataPropagationRequest{TaskQueueMaxVersions: syncRes.TaskQueueMaxVersions}
+		err = workflow.ExecuteActivity(activityCtx, d.a.CheckUserDataPropagation, checkReq).Get(ctx, nil)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *DeploymentWorkflowRunner) syncToTaskQueues2(ctx workflow.Context) error {
+	const batchSize = 100
+	type nameDataPair struct {
+		name string
+		data *deploymentspb.TaskQueueData
+	}
+
+	activityCtx := workflow.WithActivityOptions(ctx, defaultActivityOptions)
+	tqNames := workflow.DeterministicKeys(d.State.TaskQueueFamilies)
+
+	for _, tqType := range []enumspb.TaskQueueType{
+		// Sync and wait for workflow task queues, then activity task queues, to ensure that
+		// workflows don't bounce between deployments due to activity/wft starts.
+		enumspb.TASK_QUEUE_TYPE_WORKFLOW,
+		enumspb.TASK_QUEUE_TYPE_ACTIVITY,
```

Comment:
We want to support Nexus tasks too. they should be synced with activities I believe.